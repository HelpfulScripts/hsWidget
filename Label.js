import { m } from 'hslayout';
export class Label {
    view(node) {
        const css = node.attrs.css || '';
        const style = node.attrs.style || '';
        const text = node.attrs.text || 'unspecified';
        return m(`.hs-label ${css}`, { style: style }, m.trust(text));
    }
}
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiTGFiZWwuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvTGFiZWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBZ0NBLE9BQU8sRUFBRSxDQUFDLEVBQVMsTUFBVSxVQUFVLENBQUM7QUFZeEMsTUFBTTtJQUNGLElBQUksQ0FBQyxJQUFXO1FBQ1osTUFBTSxHQUFHLEdBQUcsSUFBSSxDQUFDLEtBQUssQ0FBQyxHQUFHLElBQUksRUFBRSxDQUFDO1FBQ2pDLE1BQU0sS0FBSyxHQUFHLElBQUksQ0FBQyxLQUFLLENBQUMsS0FBSyxJQUFJLEVBQUUsQ0FBQztRQUNyQyxNQUFNLElBQUksR0FBSSxJQUFJLENBQUMsS0FBSyxDQUFDLElBQUksSUFBSSxhQUFhLENBQUM7UUFDL0MsT0FBTyxDQUFDLENBQUMsYUFBYSxHQUFHLEVBQUUsRUFBRSxFQUFFLEtBQUssRUFBQyxLQUFLLEVBQUUsRUFBRSxDQUFDLENBQUMsS0FBSyxDQUFDLElBQUksQ0FBQyxDQUFDLENBQUM7SUFDakUsQ0FBQztDQUNKIn0=