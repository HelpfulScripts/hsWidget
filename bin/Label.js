"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const hslayout_1 = require("hslayout");
class Label {
    view(node) {
        const css = node.attrs.css || '';
        const style = node.attrs.style || '';
        const text = node.attrs.text || 'unspecified';
        return hslayout_1.m(`.hs-label ${css}`, { style: style }, hslayout_1.m.trust(text));
    }
}
exports.Label = Label;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiTGFiZWwuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvTGFiZWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFnQ0EsdUNBQXdDO0FBWXhDO0lBQ0ksSUFBSSxDQUFDLElBQVc7UUFDWixNQUFNLEdBQUcsR0FBRyxJQUFJLENBQUMsS0FBSyxDQUFDLEdBQUcsSUFBSSxFQUFFLENBQUM7UUFDakMsTUFBTSxLQUFLLEdBQUcsSUFBSSxDQUFDLEtBQUssQ0FBQyxLQUFLLElBQUksRUFBRSxDQUFDO1FBQ3JDLE1BQU0sSUFBSSxHQUFJLElBQUksQ0FBQyxLQUFLLENBQUMsSUFBSSxJQUFJLGFBQWEsQ0FBQztRQUMvQyxPQUFPLFlBQUMsQ0FBQyxhQUFhLEdBQUcsRUFBRSxFQUFFLEVBQUUsS0FBSyxFQUFDLEtBQUssRUFBRSxFQUFFLFlBQUMsQ0FBQyxLQUFLLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQztJQUNqRSxDQUFDO0NBQ0o7QUFQRCxzQkFPQyJ9