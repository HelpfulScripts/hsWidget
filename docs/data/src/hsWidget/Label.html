
                <html><style>body { overflow:hidden;}
            h1 { font-family: Arial, sans-serif; font-size: 24px; color: #44a; }
            p { margin:0; padding:0; }
            br  { margin:0; padding:0; }
            .line { 
                white-space: nowrap;
                height:16px; 
            }
            .line>span { 
                display:inline-block; background-color:#eef; height:100%; 
                margin: 0 5px 0 0; padding-right: 5px; color:#999;   
            }
            comment {color: #080;} module {color: #804;} 
            quote {color: #008;} comment>quote {color: #080;}
            .listing { margin: 10px; border: 1px solid #ccc; 
                    font-family: SFMono-Regular, Consolas, 'Liberation Mono', Menlo, Courier, monospace;
                    font-size: 14px;  
                    overflow:scroll;
                    height:90%;
            }
            code { padding: 5px 0;}</style></html><body>
                <h1>Label.ts</h1>
                <div class='listing'><code><p id=1 class="line"><span>&nbsp;&nbsp;&nbsp;1</span><comment>/**</comment></p>
<p id=2 class="line"><span>&nbsp;&nbsp;&nbsp;2</span><comment>&nbsp;*&nbsp;#&nbsp;Label&nbsp;Widget</comment></p>
<p id=3 class="line"><span>&nbsp;&nbsp;&nbsp;3</span><comment>&nbsp;*&nbsp;Shows&nbsp;a&nbsp;text&nbsp;label</comment></p>
<p id=4 class="line"><span>&nbsp;&nbsp;&nbsp;4</span><comment>&nbsp;*&nbsp;</comment></p>
<p id=5 class="line"><span>&nbsp;&nbsp;&nbsp;5</span><comment>&nbsp;*&nbsp;###&nbsp;Profile</comment></p>
<p id=6 class="line"><span>&nbsp;&nbsp;&nbsp;6</span><comment>&nbsp;*&nbsp;invoked&nbsp;as&nbsp;`m(Label,&nbsp;{&nbsp;&lt;LabelAttrs&gt;&nbsp;},&nbsp;&lt;content&gt;)`.</comment></p>
<p id=7 class="line"><span>&nbsp;&nbsp;&nbsp;7</span><comment>&nbsp;*&nbsp;</comment></p>
<p id=8 class="line"><span>&nbsp;&nbsp;&nbsp;8</span><comment>&nbsp;*&nbsp;See&nbsp;{@link&nbsp;Label.LabelAttrs&nbsp;LabelAttrs}</comment></p>
<p id=9 class="line"><span>&nbsp;&nbsp;&nbsp;9</span><comment>&nbsp;*&nbsp;`&lt;content&gt;`:&nbsp;the&nbsp;label&nbsp;content;&nbsp;supports&nbsp;embedded&nbsp;HTML&nbsp;code.</comment></p>
<p id=10 class="line"><span>&nbsp;&nbsp;10</span><comment>&nbsp;*&nbsp;</comment></p>
<p id=11 class="line"><span>&nbsp;&nbsp;11</span><comment>&nbsp;*&nbsp;###&nbsp;Example</comment></p>
<p id=12 class="line"><span>&nbsp;&nbsp;12</span><comment>&nbsp;*&nbsp;&lt;example&gt;</comment></p>
<p id=13 class="line"><span>&nbsp;&nbsp;13</span><comment>&nbsp;*&nbsp;&lt;file&nbsp;name=<quote>'script.js'</quote>&gt;&nbsp;</comment></p>
<p id=14 class="line"><span>&nbsp;&nbsp;14</span><comment>&nbsp;*&nbsp;m.mount(root,&nbsp;{view:&nbsp;()&nbsp;=&gt;&nbsp;m(hsWidget.Label,&nbsp;{</comment></p>
<p id=15 class="line"><span>&nbsp;&nbsp;15</span><comment>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;class:&nbsp;<quote>'myLabel'</quote>,&nbsp;</comment></p>
<p id=16 class="line"><span>&nbsp;&nbsp;16</span><comment>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;style:&nbsp;<quote>'text-align:&nbsp;right;'</quote>,</comment></p>
<p id=17 class="line"><span>&nbsp;&nbsp;17</span><comment>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;},&nbsp;<quote>'This&nbsp;is&nbsp;a&nbsp;&lt;b&gt;label&lt;/b&gt;'</quote>)</comment></p>
<p id=18 class="line"><span>&nbsp;&nbsp;18</span><comment>&nbsp;*&nbsp;});</comment></p>
<p id=19 class="line"><span>&nbsp;&nbsp;19</span><comment>&nbsp;*&nbsp;&lt;/file&gt;</comment></p>
<p id=20 class="line"><span>&nbsp;&nbsp;20</span><comment>&nbsp;*&nbsp;&lt;file&nbsp;name=<quote>'style.css'</quote>&gt;</comment></p>
<p id=21 class="line"><span>&nbsp;&nbsp;21</span><comment>&nbsp;*&nbsp;.myLabel&nbsp;{&nbsp;&nbsp;</comment></p>
<p id=22 class="line"><span>&nbsp;&nbsp;22</span><comment>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;background-color:&nbsp;white;&nbsp;</comment></p>
<p id=23 class="line"><span>&nbsp;&nbsp;23</span><comment>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;margin:&nbsp;5px;</comment></p>
<p id=24 class="line"><span>&nbsp;&nbsp;24</span><comment>&nbsp;*&nbsp;}</comment></p>
<p id=25 class="line"><span>&nbsp;&nbsp;25</span><comment>&nbsp;*&nbsp;&lt;/file&gt;</comment></p>
<p id=26 class="line"><span>&nbsp;&nbsp;26</span><comment>&nbsp;*&nbsp;&lt;/example&gt;</comment></p>
<p id=27 class="line"><span>&nbsp;&nbsp;27</span><comment>&nbsp;*&nbsp;</comment></p>
<p id=28 class="line"><span>&nbsp;&nbsp;28</span><comment>&nbsp;*/</comment></p>
<p id=29 class="line"><span>&nbsp;&nbsp;29</span></p>
<p id=30 class="line"><span>&nbsp;&nbsp;30</span><comment>/**&nbsp;*/</comment></p>
<p id=31 class="line"><span>&nbsp;&nbsp;31</span>import&nbsp;m&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;<quote>"mithril"</quote>;</p>
<p id=32 class="line"><span>&nbsp;&nbsp;32</span>import&nbsp;{&nbsp;Widget,&nbsp;ViewResult&nbsp;}&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;<quote>"./Widget"</quote>;</p>
<p id=33 class="line"><span>&nbsp;&nbsp;33</span>import&nbsp;{&nbsp;WidgetAttrs&nbsp;}&nbsp;&nbsp;from&nbsp;<quote>"./Widget"</quote>;</p>
<p id=34 class="line"><span>&nbsp;&nbsp;34</span>import&nbsp;{&nbsp;Vnode&nbsp;}&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;<quote>"./Widget"</quote>;</p>
<p id=35 class="line"><span>&nbsp;&nbsp;35</span></p>
<p id=36 class="line"><span>&nbsp;&nbsp;36</span></p>
<p id=37 class="line"><span>&nbsp;&nbsp;37</span>export&nbsp;interface&nbsp;LabelAttrs&nbsp;extends&nbsp;WidgetAttrs&nbsp;{</p>
<p id=38 class="line"><span>&nbsp;&nbsp;38</span>}</p>
<p id=39 class="line"><span>&nbsp;&nbsp;39</span></p>
<p id=40 class="line"><span>&nbsp;&nbsp;40</span><comment>/**</comment></p>
<p id=41 class="line"><span>&nbsp;&nbsp;41</span><comment>&nbsp;*&nbsp;#&nbsp;Label&nbsp;</comment></p>
<p id=42 class="line"><span>&nbsp;&nbsp;42</span><comment>&nbsp;*&nbsp;A&nbsp;text&nbsp;label</comment></p>
<p id=43 class="line"><span>&nbsp;&nbsp;43</span><comment>&nbsp;*/</comment></p>
<p id=44 class="line"><span>&nbsp;&nbsp;44</span>export&nbsp;class&nbsp;Label&nbsp;extends&nbsp;Widget&nbsp;{</p>
<p id=45 class="line"><span>&nbsp;&nbsp;45</span>&nbsp;&nbsp;&nbsp;&nbsp;view(node:&nbsp;Vnode&lt;LabelAttrs,&nbsp;this&gt;):ViewResult&nbsp;{&nbsp;</p>
<p id=46 class="line"><span>&nbsp;&nbsp;46</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;const&nbsp;text&nbsp;&nbsp;=&nbsp;&lt;string&gt;node.children[0]&nbsp;||&nbsp;<quote>''</quote>;</p>
<p id=47 class="line"><span>&nbsp;&nbsp;47</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;m(`.hs_label`,&nbsp;this.attrs(node.attrs),&nbsp;m.trust(text));&nbsp;</p>
<p id=48 class="line"><span>&nbsp;&nbsp;48</span>&nbsp;&nbsp;&nbsp;&nbsp;}</p>
<p id=49 class="line"><span>&nbsp;&nbsp;49</span>}</p>
<p id=50 class="line"><span>&nbsp;&nbsp;50</span></p>
<p id=51 class="line"><span>&nbsp;&nbsp;51</span></p></code></div>
                </body>
            