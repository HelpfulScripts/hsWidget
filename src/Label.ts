/**
 * # Label Widget
 * Shows a text label
 * 
 * ### Profile
 * invoked as `m(Label, { <LabelAttrs> }, <content>)`.
 * 
 * See {@link Label.LabelAttrs LabelAttrs}
 * `<content>`: the label content; supports embedded HTML code.
 * 
 * ### Example
 * <example>
 * <file name='script.js'> 
 * m.mount(root, {view: () => m(hsWidget.Label, {
 *      class: 'myLabel', 
 *      style: 'text-align: right;',
 *    }, 'This is a <b>label</b>')
 * });
 * </file>
 * <file name='style.css'>
 * .myLabel {  
 *    background-color: white; 
 *    margin: 5px;
 * }
 * </file>
 * </example>
 * 
 */

/** */
import m                from "mithril";
import { Widget, ViewResult }       from "./Widget";
import { WidgetAttrs }  from "./Widget";
import { Vnode }        from "./Widget";


export interface LabelAttrs extends WidgetAttrs {
}

/**
 * # Label 
 * A text label
 */
export class Label extends Widget {
    view(node: Vnode<LabelAttrs, this>):ViewResult { 
        const text  = <string>node.children[0] || '';
        return m(`.hs_label`, this.attrs(node.attrs), m.trust(text)); 
    }
}

