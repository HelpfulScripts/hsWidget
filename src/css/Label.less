@LabelHeight: 30px;

.hs-label {
    position: relative;
    margin:  0px;
    box-sizing: border-box;
    text-align: center;
    vertical-align: middle;
    margin-left:    5px;
    margin-right:   5px;
    line-height: @LabelHeight;
}