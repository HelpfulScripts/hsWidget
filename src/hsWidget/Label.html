
                <html><style>body { overflow:hidden;}
            h1 { font-family: Arial, sans-serif; font-size: 24px; color: #44a; }
            p { margin:0; padding-top:5px; }
            br  { margin:0; padding:0; }
            .line { margin: 0 5px 0 0; padding-right: 5px; color:#999; background-color:#eef;  }
            comment { color: #080;} module { color: #804;}
            .listing { margin: 10px; border: 1px solid #ccc; 
                    font-family: SFMono-Regular, Consolas, 'Liberation Mono', Menlo, Courier, monospace;
                    font-size: 14px; line-height: 1.2em; 
                    overflow:scroll;
                    height:90%;
            }
            code { padding: 5px 0;}</style></html><body>
                <h1>src/Label.ts</h1>
                <div class='listing'><code><span id=1 class="line">&nbsp;&nbsp;&nbsp;1</span><comment>/**<br>
<span id=2 class="line">&nbsp;&nbsp;&nbsp;2</span>&nbsp;*&nbsp;#&nbsp;Label&nbsp;Widget<br>
<span id=3 class="line">&nbsp;&nbsp;&nbsp;3</span>&nbsp;*&nbsp;Shows&nbsp;a&nbsp;text&nbsp;label<br>
<span id=4 class="line">&nbsp;&nbsp;&nbsp;4</span>&nbsp;*&nbsp;<br>
<span id=5 class="line">&nbsp;&nbsp;&nbsp;5</span>&nbsp;*&nbsp;###&nbsp;Profile<br>
<span id=6 class="line">&nbsp;&nbsp;&nbsp;6</span>&nbsp;*&nbsp;invoked&nbsp;as&nbsp;`m(Label,&nbsp;{text:<string>});`<br>
<span id=7 class="line">&nbsp;&nbsp;&nbsp;7</span>&nbsp;*&nbsp;<br>
<span id=8 class="line">&nbsp;&nbsp;&nbsp;8</span>&nbsp;*&nbsp;###&nbsp;Attributes&nbsp;(node.attrs):<br>
<span id=9 class="line">&nbsp;&nbsp;&nbsp;9</span>&nbsp;*&nbsp;-&nbsp;`name:&nbsp;string`&nbsp;name&nbsp;to&nbsp;show&nbsp;as&nbsp;button&nbsp;text<br>
<span id=10 class="line">&nbsp;&nbsp;10</span>&nbsp;*&nbsp;-&nbsp;`css:&nbsp;string`&nbsp;css&nbsp;class&nbsp;to&nbsp;assign&nbsp;to&nbsp;button&nbsp;tag<br>
<span id=11 class="line">&nbsp;&nbsp;11</span>&nbsp;*&nbsp;-&nbsp;`style:&nbsp;string`&nbsp;style&nbsp;string&nbsp;to&nbsp;apply&nbsp;to&nbsp;button&nbsp;tag<br>
<span id=12 class="line">&nbsp;&nbsp;12</span>&nbsp;*&nbsp;<br>
<span id=13 class="line">&nbsp;&nbsp;13</span>&nbsp;*&nbsp;###&nbsp;Example<br>
<span id=14 class="line">&nbsp;&nbsp;14</span>&nbsp;*&nbsp;<example><br>
<span id=15 class="line">&nbsp;&nbsp;15</span>&nbsp;*&nbsp;<file&nbsp;name='script.js'>&nbsp;<br>
<span id=16 class="line">&nbsp;&nbsp;16</span>&nbsp;*&nbsp;m.mount(root,&nbsp;{view:&nbsp;()&nbsp;=>&nbsp;m(hswidget.Label,&nbsp;{<br>
<span id=17 class="line">&nbsp;&nbsp;17</span>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;css:&nbsp;'.myLabel',&nbsp;<br>
<span id=18 class="line">&nbsp;&nbsp;18</span>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;style:&nbsp;'text-align:&nbsp;right;',<br>
<span id=19 class="line">&nbsp;&nbsp;19</span>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;text:&nbsp;'This&nbsp;is&nbsp;a&nbsp;<b>label</b>'<br>
<span id=20 class="line">&nbsp;&nbsp;20</span>&nbsp;*&nbsp;})});<br>
<span id=21 class="line">&nbsp;&nbsp;21</span>&nbsp;*&nbsp;</file><br>
<span id=22 class="line">&nbsp;&nbsp;22</span>&nbsp;*&nbsp;<file&nbsp;name='style.css'><br>
<span id=23 class="line">&nbsp;&nbsp;23</span>&nbsp;*&nbsp;.myLabel&nbsp;{&nbsp;&nbsp;<br>
<span id=24 class="line">&nbsp;&nbsp;24</span>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;background-color:&nbsp;white;&nbsp;<br>
<span id=25 class="line">&nbsp;&nbsp;25</span>&nbsp;*&nbsp;&nbsp;&nbsp;&nbsp;margin:&nbsp;5px;<br>
<span id=26 class="line">&nbsp;&nbsp;26</span>&nbsp;*&nbsp;}<br>
<span id=27 class="line">&nbsp;&nbsp;27</span>&nbsp;*&nbsp;</file><br>
<span id=28 class="line">&nbsp;&nbsp;28</span>&nbsp;*&nbsp;</example><br>
<span id=29 class="line">&nbsp;&nbsp;29</span>&nbsp;*&nbsp;<br>
<span id=30 class="line">&nbsp;&nbsp;30</span>&nbsp;*/</comment><br>
<span id=31 class="line">&nbsp;&nbsp;31</span><br>
<span id=32 class="line">&nbsp;&nbsp;32</span><comment>/**&nbsp;*/</comment><br>
<span id=33 class="line">&nbsp;&nbsp;33</span>import&nbsp;{&nbsp;m,&nbsp;Vnode&nbsp;}&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;'hslayout';<br>
<span id=34 class="line">&nbsp;&nbsp;34</span><br>
<span id=35 class="line">&nbsp;&nbsp;35</span><comment>/**<br>
<span id=36 class="line">&nbsp;&nbsp;36</span>&nbsp;*&nbsp;#&nbsp;Label&nbsp;<br>
<span id=37 class="line">&nbsp;&nbsp;37</span>&nbsp;*&nbsp;A&nbsp;text&nbsp;label<br>
<span id=38 class="line">&nbsp;&nbsp;38</span>&nbsp;*&nbsp;<br>
<span id=39 class="line">&nbsp;&nbsp;39</span>&nbsp;*&nbsp;###&nbsp;Profile<br>
<span id=40 class="line">&nbsp;&nbsp;40</span>&nbsp;*&nbsp;invoked&nbsp;as&nbsp;`m(Label,&nbsp;{text:<string>});`<br>
<span id=41 class="line">&nbsp;&nbsp;41</span>&nbsp;*&nbsp;<br>
<span id=42 class="line">&nbsp;&nbsp;42</span>&nbsp;*&nbsp;###&nbsp;Attributes&nbsp;(node.attrs):<br>
<span id=43 class="line">&nbsp;&nbsp;43</span>&nbsp;*&nbsp;-&nbsp;`text?:&nbsp;string`&nbsp;the&nbsp;label&nbsp;text;&nbsp;<br>
<span id=44 class="line">&nbsp;&nbsp;44</span>&nbsp;*/</comment><br>
<span id=45 class="line">&nbsp;&nbsp;45</span>export&nbsp;class&nbsp;Label&nbsp;{<br>
<span id=46 class="line">&nbsp;&nbsp;46</span>&nbsp;&nbsp;&nbsp;&nbsp;view(node:&nbsp;Vnode):&nbsp;Vnode&nbsp;{&nbsp;<br>
<span id=47 class="line">&nbsp;&nbsp;47</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;const&nbsp;css&nbsp;=&nbsp;node.attrs.css&nbsp;||&nbsp;'';<br>
<span id=48 class="line">&nbsp;&nbsp;48</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;const&nbsp;style&nbsp;=&nbsp;node.attrs.style&nbsp;||&nbsp;'';<br>
<span id=49 class="line">&nbsp;&nbsp;49</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;const&nbsp;text&nbsp;&nbsp;=&nbsp;node.attrs.text&nbsp;||&nbsp;'unspecified';<br>
<span id=50 class="line">&nbsp;&nbsp;50</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;m(`.hs-label&nbsp;${css}`,&nbsp;{&nbsp;style:style&nbsp;},&nbsp;m.trust(text));&nbsp;<br>
<span id=51 class="line">&nbsp;&nbsp;51</span>&nbsp;&nbsp;&nbsp;&nbsp;}<br>
<span id=52 class="line">&nbsp;&nbsp;52</span>}<br>
<span id=53 class="line">&nbsp;&nbsp;53</span><br>
<span id=54 class="line">&nbsp;&nbsp;54</span><br></code></div>
                </body>
            