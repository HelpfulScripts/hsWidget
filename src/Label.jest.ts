// import * as hswidget from './';
import { Label }    from './Label';
import m from "mithril";

const root = window.document.createElement("div");

describe('Label', () => {
    describe('should have DOM structure', () => {
        beforeAll(() => {
            m.mount(root, { view: () => m(Label, {
                class: 'myLabel',
                style: 'text-align: right;',
            }, 'This is a label')});
        });
        it ('matches', () => {
            expect(root).toMatchSnapshot();
        });
    });
    describe('should work with defaults', () => {
        beforeAll(() => {
            m.mount(root, { view: () => m(Label)});
        });
        it ('matches', () => {
            expect(root).toMatchSnapshot();
        });
    });
});