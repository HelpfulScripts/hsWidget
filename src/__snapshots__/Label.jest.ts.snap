// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should have DOM structure matches 1`] = `
<div>
  <div
    class="hs_label  myLabel"
    style="text-align: right;"
  >
    This is a label
  </div>
</div>
`;

exports[`Label should work with defaults matches 1`] = `
<div>
  <div
    class="hs_label"
  />
</div>
`;
